### PR TITLE
Chef 17 updates

### LINKED ISSUE #449: percona Chef 17 compatibility

### PR DESCRIPTION
**Pull request by ramereth**
### Description

- Chef 17 updates: enable `unified_mode` on all resources
- Remove dependency on openssl cookbook and create `percona_secure_random` method to replace that functionality
- Remove dependency on chef-vault cookbook and require Chef >= 16 

### Issues Resolved

Closes #449 

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
This supersedes #446

**Comment by damacus** [comment]
We should need, or require, Chef-Vault for this cookbook to work. I'm wondering if we need to pause this and fix those things first?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,13 +41,10 @@ jobs:
       matrix:
         os:
           - 'debian-9'
-          - 'debian-9-chef-13'
           - 'debian-10'
           - 'centos-7'
-          - 'centos-7-chef-13'
           - 'centos-8'
```
Comment (damacus) [inline .github/workflows/ci.yml:46]: Do you also want stream?

Comment (ramereth) [reply .github/workflows/ci.yml:46]: Probably wouldn't hurt to add it. I'll do that.

**Comment by ramereth** [comment]
> We should need, or require, Chef-Vault for this cookbook to work. I'm wondering if we need to pause this and fix those things first?

I feel as though we may want to look at removing the reliance to the Chef-Vault cookbook somehow to get this release to work. I set this as a draft PR since it has some deps.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ramereth) [reply .github/workflows/ci.yml:46]: I'll do that in a different PR. I want to just get this out in case I run into issues on stream.

**Comment by ramereth** [comment]
Supersedes #452

File: README.md [hunk 2]
```diff
@@ -34,14 +34,12 @@ We provide an expanding set of tests against the following 64-bit platforms whic
...
 
-- [openssl](https://supermarket.chef.io/cookbooks/openssl)
 - [yum-epel](https://supermarket.chef.io/cookbooks/yum-epel)
-- [chef-vault](https://supermarket.chef.io/cookbooks/chef-vault)
 - [line](https://supermarket.chef.io/cookbooks/line)
 
 ### Chef
 
-This cookbook requires Chef >= 13.
+This cookbook requires Chef >= 15.3.
```
Comment (bmhughes) [inline README.md:42]: This should be >= 16 shouldn't it?

Comment (ramereth) [reply README.md:42]: Nice catch. I'll fix that.

**Review by bmhughes — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [3.0.0](https://github.com/sous-chefs/percona/releases/tag/3.0.0)

**Pull request merged**